### PR TITLE
78 refactor land use map

### PR DESCRIPTION
**Pull request by chriswhong**
This PR:
- Overrides `mapbox-gl-source` and `mapbox-gl-layer` components from `ember-mapbox-gl`  to allow for parity with the MapboxGL JS api.
- Refactors all GL-based map views to use the new components
- Refactors land use map to show raster tiles below zoom level 14, and vector tiles with tooltips at 14 and above
- Consumes pluto/land use vector and raster tiles from the new `tiles.planninglabs.nyc`

### CODE REVIEW

File: app/components/land-use-map.js [hunk 1]
```diff
@@ -1,12 +1,110 @@
...
-    dashArray: '8 5',
-    opacity: 0.6,
+  initOptions: {
+    style: 'mapbox://styles/mapbox/light-v9',
+    zoom: 9,
+    center: [-74, 40.7071],
+    scrollZoom: false,
   },
+
+  vectorSource: Ember.computed('mapState.landUseTemplate', function () {
```
Comment (allthesignals) [inline app/components/land-use-map.js]: I don't think these need to be computed properties if they're not doing anything aside from returning object literals.

File: app/components/land-use-map.js [hunk 2]
```diff
@@ -1,12 +1,110 @@
...
+    return {
+      type: 'vector',
+      tiles: ['https://tiles.planninglabs.nyc/pluto/{z}/{x}/{y}/tile.mvt'],
+      minzoom: 14,
+    };
+  }),
+
+  rasterSource: Ember.computed('mapState.landUseTemplate', function () {
```
Comment (allthesignals) [inline app/components/land-use-map.js]: I don't think these need to be computed properties if they're not doing anything aside from returning object literals.

File: app/components/land-use-map.js (hunk 1 above)
Comment (chriswhong) [inline app/components/land-use-map.js]: Good catch...  they used to be dependent on an async call to carto, now the URLs are static.

File: app/components/land-use-map.js (hunk 2 above)
Comment (chriswhong) [inline app/components/land-use-map.js]: Fixed in 1c0733e75fb09e9b5942acc52ea612b28fb2140f

File: app/templates/application.hbs [hunk 3]
```diff
@@ -25,32 +25,21 @@
...
-        {{#map.source data=mapState.geom sourceId='currentlySelected' as |source|}}
-          {{source.layer  layerType='fill'
-                          source='currentlySelected'
-                          paintOptions=(hash fill-color='#ae561f' fill-opacity=0.2 fill-outline-color='#ae561f')}}
+        {{#map.source sourceId='currentlySelected' source=cdSelectedSource as |source|}}
+          {{source.layer layer=cdSelectedLayer}}
         {{/map.source}}
       {{/if}}
-
-      {{#if mouseoverLocation}}
```
Comment (allthesignals) [inline app/templates/application.hbs]: @chriswhong want to make sure removing tooltip was intentional - I remember some discussion about how it was redundant with the labels.

File: app/templates/components/land-use-map.hbs [hunk 4]
```diff
@@ -1,8 +1,28 @@
-{{#leaflet-map bounds=mapState.bounds scrollWheelZoom=false minZoom=14 classNames='land-use-map'}}
-  {{tile-layer url="http://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png"}}
-  {{#if mapState.landUseTemplate}}
-    {{tile-layer url=mapState.landUseTemplate}}
+{{#mapbox-gl id='lu-map' initOptions=initOptions as |map|}}
```
Comment (allthesignals) [inline app/templates/components/land-use-map.hbs]: For later, but this map probably needs a zoom control

**Pull request merged**